File: src/main/java/de/bettinggame/domain/News.java
package de.bettinggame.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import java.time.OffsetDateTime;

/**
 * News entity.
 */
@Entity
@Table(name = "news")
public class News extends AbstractIdEntity {

    @NotNull
    @Column(nullable = false)
    private String headline;

    @NotNull
    @Column(nullable = false)
    private String content;

    @NotNull
    @Column(name = "publish_date", nullable = false)
    private OffsetDateTime publishDate;

    @NotNull
    @Column(name="author_username", nullable = false)
    private String authorUsername;

    @ManyToOne
    @JoinColumn(name = "author_id", referencedColumnName = "id")
    private User author;

    protected News() {
    }

    public String getHeadline() {
        return headline;
    }

    public String getContent() {
        return content;
    }

    public OffsetDateTime getPublishDate() {
        return publishDate;
    }

    public String getAuthorUsername() {
        return authorUsername;
    }

    public String getAuthor() {
        if (author == null) {
            return authorUsername;
        }
        return author.getFullname().orElse(authorUsername);
    }
}

File: src/main/kotlin/de/bettinggame/domain/News.kt
package de.bettinggame.domain

import org.springframework.data.jpa.repository.JpaRepository
import java.time.OffsetDateTime
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.validation.constraints.NotNull

/**
 * News entity.
 */
@Entity
class News(identifier: String,
           @NotNull
           val headline: String,
           @NotNull
           val content: String,
           @NotNull
           @Column(name = "publish_date", nullable = false)
           val publishDate: OffsetDateTime,
           @NotNull
           val authorUsername: String,
           @ManyToOne
           @JoinColumn(name = "author_id", referencedColumnName = "id")
           var author: User?
) : AbstractIdentifiableEntity(identifier) {
    val authorName: String?
        get() {
            if (author == null) {
                return authorUsername
            }
            return author?.fullname ?: authorUsername
        }
}

/**
 * News repository.
 */
interface NewsRepository : JpaRepository<News, Int> {
    fun findAllByOrderByPublishDateDesc(): List<News>
}
